v2/cmd/asoctl/pkg/importing/resource_importer.go

```diff
@@ -243,39 +239,36 @@ func (ri *ResourceImporter) collateResults(
 				"name", rsrc.Name())
 
 			report.AddSuccessfulImport(rsrc)
-			ri.imported[rsrc.Id()] = rsrc
+			ri.imported[rsrc.ID()] = rsrc
 		}
 
 		// Flag the main resource as complete
 		// We do this after everything else because it might indicate we're finished
-		progress <- progressDelta{complete: 1}
+		progress.Completed(1)
 	}
 
-	report.WriteToLog(ri.log)
+	publish <- report
 }
 
 func (ri *ResourceImporter) ImportResource(
```

Comment:
Do we want this exported? Or is the intent that the user always should use the `Add` + `Import` combination, in which case this could be made private?

Reply:
Made this private.